packages/node-cli/src/commands/ManageAccountCommandGenerator.js

```diff
@@ -0,0 +1,284 @@
+				},
+				{
+					name: NodeTranslationService.getMessage(QUESTIONS_CHOICES.ACTIONS.REMOVE),
+					value: ACTION.REMOVE,
+				},
+				// {
+				//    name: NodeTranslationService.getMessage(QUESTIONS_CHOICES.ACTIONS.REVOKE),
+				//    value: ACTION.REVOKE,
+				// },
+				{
+					name: NodeTranslationService.getMessage(QUESTIONS_CHOICES.ACTIONS.EXIT),
+					value: ACTION.EXIT,
+				},
+			],
+		});
+
+		if (answer[ANSWERS_NAMES.ACTION] == ACTION.EXIT) {
+			throw NodeTranslationService.getMessage(MESSAGES.CANCEL);
+		}
+
+		return answer[ANSWERS_NAMES.ACTION];
+	}
+
+	async _introduceNewName(prompt, authIDList, originalAuthId) {
+		const answer = await prompt({
+			type: CommandUtils.INQUIRER_TYPES.INPUT,
+			name: ANSWERS_NAMES.RENAMETO,
+			message: NodeTranslationService.getMessage(QUESTIONS.NEW_NAME),
+			filter: (fieldValue) => fieldValue.trim(),
+			validate: (fieldValue) =>
+				showValidationResults(
+					fieldValue,
+					validateFieldIsNotEmpty,
+					validateFieldHasNoSpaces,
+					(fieldValue) => validateSameAuthID(fieldValue, originalAuthId),
+					(fieldValue) => validateAuthIDNotInList(fieldValue, Object.keys(authIDList)),
+					validateAlphanumericHyphenUnderscore,
+					validateMaximumLength
+				),
+		});
+		return answer[ANSWERS_NAMES.RENAMETO];
+	}
+
+	async _confirmRemove(prompt) {
+		const answer = await prompt([
+			{
+				type: CommandUtils.INQUIRER_TYPES.LIST,
+				name: ANSWERS_NAMES.REMOVE,
+				message: NodeTranslationService.getMessage(QUESTIONS.VERIFY_REMOVE),
+				default: false,
+				choices: [
+					{ name: NodeTranslationService.getMessage(YES), value: true },
+					{ name: NodeTranslationService.getMessage(NO), value: false },
+				],
+			},
+		]);
+		if (!answer[ANSWERS_NAMES.REMOVE]) {
+			throw NodeTranslationService.getMessage(MESSAGES.CANCEL);
+		}
+		return answer[ANSWERS_NAMES.REMOVE];
+	}
+
+	_extractAnswers(answers) {
+		if (answers[ANSWERS_NAMES.ACTION] == ACTION.RENAME) {
+			return {
+				[COMMAND.OPTIONS.RENAME]: answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId,
+				[COMMAND.OPTIONS.RENAMETO]: answers[ANSWERS_NAMES.RENAMETO],
+			};
+		} else if (answers[ANSWERS_NAMES.ACTION] == ACTION.REMOVE) {
+			return {
+				[COMMAND.OPTIONS.REMOVE]: answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId,
+			};
+			// } else if (answers[ANSWERS_NAMES.ACTION] == ACTION.REVOKE) {
+			//    return {
+			//    [COMMAND.OPTIONS.REVOKE]: answers[ANSWERS_NAMES.SELECTED_AUTH_ID].authId,
+			// }
+		}
+	}
+
+	async _executeAction(answers) {
+		const sdkParams = CommandUtils.extractCommandOptions(answers, this._commandMetadata);
+		const flags = [];
+		if (answers[COMMAND.OPTIONS.LIST]) {
+			flags.push(COMMAND.OPTIONS.LIST);
+			delete sdkParams[COMMAND.OPTIONS.LIST];
+		}
+		const executionContext = new SdkExecutionContext({
+			command: this._commandMetadata.sdkCommand,
+			params: sdkParams,
+			flags,
+		});
+
+		const action = this._getActionExecuted(answers);
+		const message = this._getSpinnerMessage(action);
```

Comment:
_getSpinnerMessage() needs both action and answers as parameters. And I think we don't need to store the result in a variable, we can directly call it in the executeWithSpinner.